### ISSUE TITLE
Encounter "Memory Error" when converting imagenet dataset

### ISSUE THREAD

**Issue by jeff830107** [aaron-xichen/pytorch-playground#18]
Hi,
When I was trying to using the Alexnet model, I first of all tried to follow your instruction to download val224_compressed.pkl and executed the command "python convert.py"
But when I was converting, it always come to the error message "Memory Error".
I am curious about how to deal with this issue, since I think the memory of the machine I used is big enough, which is 64 GB.
Thanks !

**Comment by gdjmck** [comment]
I ran into the same problem as well, the 244*244 file was dumped okay with 7.5G and the 299*299 pkl file was empty with 0B

**Comment by jnorwood** [comment]
I saw the same issue.  I separated the 224 and 299 dump processing loops and cleared variables that were no longer used.  Still it dies in dump_pickle, which must be making another copy.
So, I looked around and found that scikit learn has a joblib.dump that can replace pkl.dump in dump_pickle, and it doesn't use as much memory while writing out the files.
I think  you'll still need to separate the 224 and 299 processing, as mine was running out of 32G memory while doing a transpose.... too many copies of the same data going on.  With joblib, memory use goes up to 27G, and no error.   This could probably use a db, instead of all this image info in a dict.
